File: qqbot-gateway/src/main/java/com/mieai/qqbot/gateway/GatewayScheduler.java
package com.mieai.qqbot.gateway;

import java.time.Duration;

/** Scheduling boundary used for heartbeats, ACK deadlines, and reconnect delays. */
@FunctionalInterface
public interface GatewayScheduler {
    Cancellable schedule(Duration delay, Runnable task);

    @FunctionalInterface
    interface Cancellable {
        void cancel();
    }
}

File: qqbot-gateway/src/main/kotlin/com/mieai/qqbot/gateway/GatewayScheduler.kt
package com.mieai.qqbot.gateway

import java.time.Duration

/** Scheduling boundary used for heartbeats, ACK deadlines, and reconnect delays. */
fun interface GatewayScheduler {
    fun schedule(delay: Duration, task: () -> Unit): Cancellable

    fun interface Cancellable {
        fun cancel()
    }
}
